### PR TITLE
Added Harming Potion

### PR DESCRIPTION
**Pull request by camelliaguan3**
Changed a bunch of syntax errors in my project

### CODE REVIEW

File: src/main/java/org/terasology/potions/effect/HarmEffect.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import org.terasology.logic.health.DoDamageEvent;
+import org.terasology.potions.HerbEffect;
+import org.terasology.entitySystem.entity.*;
+import org.terasology.math.*;
+public class HarmEffect implements HerbEffect{
+    /**
+     * The base amount of healing allowed. It's intended to be later multiplied by the magnitude to get the final
+     * healing amount.
+     */
+    private int baseHeal = 10;
```
Comment (xrtariq2594) [inline src/main/java/org/terasology/potions/effect/HarmEffect.java:27]: Should be `baseDamage` or `baseHarm`. Make sure to change future instances of healing to damage or harm.

File: src/main/java/org/terasology/potions/effect/HarmEffect.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    /**
+     * Apply the harming effect onto the entity. This will heal or harm 100 (or baseHeal) * magnitude HP.
+     *
+     * @param instigator    The instigator who is applying this effect on the entity. It can be a herb, potion, etc.
+     * @param entity        The entity who the healing effect is being applied on.
```
Comment (xrtariq2594) [inline src/main/java/org/terasology/potions/effect/HarmEffect.java:33]: The entity who the _harm_ effect is being applied on.

File: src/main/java/org/terasology/potions/effect/HarmEffect.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+     *                      harming quantity.
+     * @param duration      The duration of the harming effect in milliseconds. It's unused here as the heal is instant.
+     */
+    @Override
+    public void applyEffect(EntityRef instigator, EntityRef entity, float magnitude, long duration) {
+        entity.send(new DoDamageEvent(TeraMath.floorToInt(baseHeal * magnitude)));
+    }
+
+    /**
+     * Apply the healing effect onto the entity. This will heal 100 (or baseHeal) * magnitude HP.
```
Comment (xrtariq2594) [inline src/main/java/org/terasology/potions/effect/HarmEffect.java:44]: Apply the _harm_ effect

File: src/main/java/org/terasology/potions/effect/HarmEffect.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+     *
+     * @param instigator    The instigator who is applying this effect on the entity. It can be a herb, potion, etc.
+     * @param entity        The entity who the healing effect is being applied on.
```
Comment (xrtariq2594) [inline src/main/java/org/terasology/potions/effect/HarmEffect.java:47]: The _harm_ effect.

File: src/main/java/org/terasology/potions/effect/HarmEffect.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+     * @param id            Unused.
+     * @param magnitude     The magnitude of the healing effect. It's multiplied by baseHeal to determine the final
```
Comment (xrtariq2594) [inline src/main/java/org/terasology/potions/effect/HarmEffect.java]: Magnitude of the harm effect.

File: src/main/java/org/terasology/potions/effect/HarmEffect.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+     *                      healing quantity.
```
Comment (xrtariq2594) [inline src/main/java/org/terasology/potions/effect/HarmEffect.java:50]: Magnitude of the _harm_ effect.

_damage_ quantity

File: src/main/java/org/terasology/potions/effect/HarmEffect.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+     * @param duration      The duration of the healing effect in milliseconds. It's unused here as the heal is instant.
```
Comment (xrtariq2594) [inline src/main/java/org/terasology/potions/effect/HarmEffect.java:51]: Duration of the _harm_ effect.

File: src/main/java/org/terasology/potions/effect/HarmEffect.java (hunk 2 above)
Comment (camelliaguan3) [reply src/main/java/org/terasology/potions/effect/HarmEffect.java:33]: oops

File: src/main/java/org/terasology/potions/effect/HarmEffect.java (hunk 3 above)
Comment (camelliaguan3) [reply src/main/java/org/terasology/potions/effect/HarmEffect.java:44]: oooops

File: src/main/java/org/terasology/potions/effect/HarmEffect.java (hunk 4 above)
Comment (camelliaguan3) [reply src/main/java/org/terasology/potions/effect/HarmEffect.java:47]: :|

File: src/main/java/org/terasology/potions/effect/HarmEffect.java (hunk 6 above)
Comment (camelliaguan3) [reply src/main/java/org/terasology/potions/effect/HarmEffect.java:50]: Oh my god I thought I changed all of those...

File: src/main/java/org/terasology/potions/effect/HarmEffect.java (hunk 7 above)
Comment (camelliaguan3) [reply src/main/java/org/terasology/potions/effect/HarmEffect.java:51]: ..

**Comment by xrtariq2594** [comment]
Still, these are mostly minor fixes. The rest appears to be okay.

**Comment by xrtariq2594** [comment]
Already merged the updated PR.

**Pull request closed without merging**